src/internal/operators/dematerialize.ts

```diff
@@ -66,12 +68,18 @@ class DeMaterializeOperator<T extends Notification<any>, R> implements Operator<
  * @ignore
  * @extends {Ignored}
  */
-class DeMaterializeSubscriber<T extends Notification<any>> extends Subscriber<T> {
-  constructor(destination: Subscriber<any>) {
+class DeMaterializeSubscriber<N extends ObservableNotification<any>> extends Subscriber<N> {
+  constructor(destination: Subscriber<GetNotificationValueOf<N>>) {
     super(destination);
   }
 
-  protected _next(value: T) {
-    value.observe(this.destination);
+  protected _next(notification: N) {
+    observeNotification(notification, this.destination);
   }
 }
+
+type GetNotificationValueOf<T> = T extends { kind: 'N'|'E'|'C' } ?
```

Comment:
The naming of this type differs from the convention used elsewhere. Other similar types are not prefixed with `Get` - see https://github.com/ReactiveX/rxjs/blob/e1d35dc258edea0237ef49a31f7b34c058755969/src/internal/types.ts#L143

Also, I think this will need to be exported, as it's in the public API.

Reply:
Renamed to `ValueFromNotification` to match things like `ValueFromArray`.